<a href="https://colab.research.google.com/github/Echevarriaj93/Data_in_Motion_weeklies/blob/main/DIMWeeklyPandas1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Data in Motion**: Weekly pandas Challenge 1

Challenge Questions
1. Which was the most-ordered item?
2. For the most-ordered item, how many items were ordered?
3. What was the most ordered item in the choice_description column?
4. How many items were ordered in total?
5. Turn the item price into a float
6. How much was the revenue for the period in the dataset?
7. How many orders were made in the period?
8. What is the average revenue amount per order?
9. How many different items are sold?


Data URL: https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv

In [4]:
#importing dependencies
import pandas as pd
import numpy as np

In [11]:
url = 'https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv'

#since the data is a tsv we must include the sep parameter in the fucntion
data = pd.read_csv(url, sep = '\t')

data.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98

In [12]:
data.shape

(4622, 5)

In [13]:
data.isnull().sum()

order_id                 0
quantity                 0
item_name                0
choice_description    1246
item_price               0
dtype: int64

## 1. Which was the most-ordered item?

In [14]:
data.item_name.value_counts()

Chicken Bowl                             726
Chicken Burrito                          553
Chips and Guacamole                      479
Steak Burrito                            368
Canned Soft Drink                        301
Steak Bowl                               211
Chips                                    211
Bottled Water                            162
Chicken Soft Tacos                       115
Chips and Fresh Tomato Salsa             110
Chicken Salad Bowl                       110
Canned Soda                              104
Side of Chips                            101
Veggie Burrito                            95
Barbacoa Burrito                          91
Veggie Bowl                               85
Carnitas Bowl                             68
Barbacoa Bowl                             66
Carnitas Burrito                          59
Steak Soft Tacos                          55
6 Pack Soft Drink                         54
Chips and Tomatillo Red Chili Salsa       48
Chicken Cr

## 2. For the most-ordered item, how many items were ordered?

### Answer: 726

## 3. What was the most ordered item in the choice_description column?

In [20]:
data[['choice_description']][data['choice_description'].isna() == False].value_counts()

choice_description                                                                                                     
[Diet Coke]                                                                                                                134
[Coke]                                                                                                                     123
[Sprite]                                                                                                                    77
[Fresh Tomato Salsa, [Rice, Black Beans, Cheese, Sour Cream, Lettuce]]                                                      42
[Fresh Tomato Salsa, [Rice, Black Beans, Cheese, Sour Cream, Guacamole, Lettuce]]                                           40
                                                                                                                          ... 
[Roasted Chili Corn Salsa, [Fajita Vegetables, Rice, Cheese, Guacamole, Lettuce]]                                     

### Answer: Diet Coke

## 4. How many items were ordered in total?

In [22]:
data['item_name'].value_counts().sum()

4622

## 5. Turn the item price into a float

In [37]:
df = data.copy()

In [38]:
 df['item_price'] = df['item_price'].str.replace('$', '').astype('float64')

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  """Entry point for launching an IPython kernel.


In [40]:
df['item_price'].dtype

dtype('float64')

## 6. How much was the revenue for the period in the dataset?

In [46]:
rev = df.item_price.sum()

rev

34500.16

## 7. How many orders were made in the period?


In [47]:
totorders = df.quantity.sum()
totorders

4972

## 8. What is the average revenue amount per order?

In [48]:
rev / totorders

6.938889782783589

## 9. How many different items are sold?

In [51]:
df.item_name.nunique()

50